src/hooks/collaboratorHooks/index.ts

```diff
@@ -0,0 +1,119 @@
+
+import { useSuccessToast, useErrorToast } from "utils/toasts"
+
+import { DEFAULT_RETRY_MSG } from "utils"
+
+export const useListCollaboratorsHook = (siteName: string) => {
+  const errorToast = useErrorToast()
+  const { setRedirectToPage } = useRedirectHook()
+  return useQuery(
+    [LIST_COLLABORATORS_KEY, siteName],
+    () => CollaboratorService.listCollaborators(siteName),
+    {
+      onError: (err: any) => {
+        if (err?.response?.status === 403) {
+          // This is to cater for the case where the user
+          // deletes themselves from the site's collaborators list
+          setRedirectToPage("/sites")
+        } else {
+          errorToast({
+            description: `The list of collaborators could not be retrieved. ${DEFAULT_RETRY_MSG}`,
+          })
+        }
+      },
+    }
+  )
+}
+
+export const useGetCollaboratorRoleHook = (siteName: string) => {
+  const errorToast = useErrorToast()
+  return useQuery(
+    [GET_COLLABORATOR_ROLE_KEY, siteName],
+    () => CollaboratorService.getRole(siteName),
+    {
+      onError: () => {
+        errorToast({
+          description: `Your collaborator role could not be retrieved. ${DEFAULT_RETRY_MSG}`,
+        })
+      },
+    }
+  )
+}
+
+export const useDeleteCollaboratorHook = (siteName: string) => {
+  const queryClient = useQueryClient()
+  const successToast = useSuccessToast()
+  const errorToast = useErrorToast()
+  return useMutation(
+    (collaboratorId: string) =>
+      CollaboratorService.deleteCollaborator(siteName, collaboratorId),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([LIST_COLLABORATORS_KEY, siteName])
+        successToast({ description: "Collaborator removed successfully" })
+      },
+      onError: (err: any) => {
+        if (err?.response?.status === 422) {
+          errorToast({
+            description: `You can't be removed, because sites need at least one Admin`,
+          })
+        } else {
+          errorToast({
+            description: `Could not delete site member. ${DEFAULT_RETRY_MSG}`,
+          })
+        }
+      },
+    }
+  )
+}
+
+export const useAddCollaboratorHook = (
+  siteName: string,
+  setAddCollaboratorError: Dispatch<SetStateAction<string>>,
+  setModalState: Dispatch<SetStateAction<CollaboratorModalState>>,
+  isAcknowledged: boolean,
+  setIsAcknowledged: Dispatch<SetStateAction<boolean>>
+) => {
+  const queryClient = useQueryClient()
+  const successToast = useSuccessToast()
+  const ACK_REQUIRED_ERROR_MESSAGE = "Acknowledgement required"
+  return useMutation(
+    (email: string) =>
+      CollaboratorService.addCollaborator(siteName, email, isAcknowledged),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([LIST_COLLABORATORS_KEY, siteName])
+        successToast({ description: "Collaborator added successfully" })
+        setAddCollaboratorError("")
+        setModalState(CollaboratorModalState.Default)
+        setIsAcknowledged(false)
```

Comment:
Thanks for pointing this out - as discussed today during Isomer sync, we'll fix this as a separate PR after our 31 Oct deadline, since it is a non-functional blocking refactor.